### PR TITLE
winbox: fix startup of the application

### PR DESCRIPTION
**Pull request by drupol**
Hello,

I can't run Winbox with the current state of things:

```
$ NIXPKGS_ALLOW_UNFREE=1 nix shell .#winbox --impure
$ winbox
/nix/store/i9zwsxk7x6962rih4xzvhqgdq37d6blq-winbox-3.37/bin/winbox: line 11: 388601 Segmentation fault      (core dumped) /nix/store/jg8ahqq8lpbsyqdqvwqvfpcrvsq4cih4-wine-wow-7.20-staging/bin/wine /nix/store/1p1jv36pgbl3ga8lvrbkkw0ff3vszw8b-winbox64.exe "$@"
```

With this PR:

```
$ NIXPKGS_ALLOW_UNFREE=1 nix shell .#winbox --impure
$ winbox
creating Window Class routeros_null
creating Window Class routeros_connect
DPI=96
EMS=13
ERROR: bad bmp format id=2329
biPlanes=1, biBitCount=1, biCompression=0
ERROR: bad bmp format id=3329
biPlanes=1, biBitCount=1, biCompression=0
creating Window Class routeros_dbl_canvas
rescaleDPI starting-dpi=96, zoom=0 result=96
 discovery started
rescaning
connecting to 192.168.2.255
connecting to 169.254.255.255
connecting to 172.20.255.255
connecting to 172.21.255.255
connecting to 172.19.255.255
connecting to 172.17.255.255
loadConfig C:\users\devlin\AppData\Roaming\Mikrotik\Winbox\Addresses.cdb
ConnectWindow::destroyed
App::saveAllSettings
App::saveAllSettings done
saving settings to .\viw318f.tmp... for C:\users\devlin\AppData\Roaming\Mikrotik\Winbox\settings.cfg.viw
settings saved
exiting
WARNING: deleting font h=0xa0a0174
~Discovery
discovery stopped
```

The app works without any troubles.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by yrd — CHANGES_REQUESTED** [review]
Good catch! Although you can use `wine.meta.mainProgram` instead:

File: pkgs/tools/admin/winbox/default.nix [hunk 1]
```diff
@@ -36,7 +39,7 @@ let
       ${wine}/bin/wineboot -u
     fi
 
-    ${wine}/bin/wine ${executable} "$@"
+    ${wine}/bin/${wineExecutable} ${executable} "$@"
```
Comment (yrd) [inline pkgs/tools/admin/winbox/default.nix:42]:
```suggestion
    ${wine}/bin/${wine.meta.mainProgram} ${executable} "$@"
```

(and drop `wineExecutable` again)

**Review by yrd — CHANGES_REQUESTED** [review]

**Comment by yrd** [comment]
And while you're at it, would you mind changing
https://github.com/NixOS/nixpkgs/blob/1aab5c848e649f0f766f9722091bc450cae3e4e3/pkgs/tools/admin/winbox/default.nix#L20
to `fetchurl (if (wine.meta.mainProgram == "wine64") then` and dropping `use64` entirely both in the package and in `all-packages.nix`? It's probably cleaner that way ^^

File: pkgs/tools/admin/winbox/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/tools/admin/winbox/default.nix]: I don't know how that thing is working, but I'm glad to have learned something new :)

Comment (drupol) [reply pkgs/tools/admin/winbox/default.nix]: I just tested everything and it works. Where is the "wine64" string is generated ? Looks like black magic to me now :D

Comment (yrd) [reply pkgs/tools/admin/winbox/default.nix]: It's from here:
https://github.com/NixOS/nixpkgs/blob/58930f0b7f3171d398fbc7b297ff0a8eb18de537/pkgs/applications/emulators/wine/base.nix#L201
Which is populated in here for wine64:
https://github.com/NixOS/nixpkgs/blob/58930f0b7f3171d398fbc7b297ff0a8eb18de537/pkgs/applications/emulators/wine/packages.nix#L44
The default being here again:
https://github.com/NixOS/nixpkgs/blob/58930f0b7f3171d398fbc7b297ff0a8eb18de537/pkgs/applications/emulators/wine/base.nix#L9

**Review by yrd — APPROVED** [review]
Awesome, thanks! 😀

**Review by yrd — APPROVED** [review]

**Pull request merged**